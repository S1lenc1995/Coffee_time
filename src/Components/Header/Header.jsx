import { useState, useEffect, useContext } from "react"; 
import PropTypes from 'prop-types';
import {Link, NavLink} from "react-router-dom"
import { ReactComponent as BasketIcon } from "./icons/basketIcon.svg";
import { ReactComponent as FavoriteIcon } from "./icons/favoriteIcon.svg";
import './Header.scss'


function Header (props) {
     const { countFavorit, countBasket } = props  
         return (
            <header className='header'>
                <div className='header__nav'>
                    <div className="header__nav--item">
                        <img src="https://static.tildacdn.com/tild3937-3261-4630-b235-343635633134/logo_white_mini.png" alt="photo" />
                        <div className='header__nav--item--icon'>
                            <div>
                                <NavLink to="/" activeClassName="active" className="icon-page">
                                <p>Main Page</p>
                                </NavLink>
                            </div>
                            <div className='header__nav--item--icon--basket'>
                                <NavLink to="/basket" activeClassName="active" className="icon-page">
                                <span className='count-basket'>{countBasket}</span>
                                <BasketIcon/>
                                </NavLink>
                            </div>
                            <div className='header__nav--item--icon--favorites'>
                                <NavLink to="/favorites" activeClassName="active" className="icon-page">
                                <span className='count-favorites'>{countFavorit}</span>
                                <FavoriteIcon/>
                                </NavLink>
                            </div>
                        </div>
                    </div>
                </div>

                <h1 className='header__name'>COFFEE <br /> TIME</h1>
            </header>
        )
    
}

 Header.propTypes = {
	countBasket: PropTypes.number,
    countFavorit: PropTypes.number
}; 


export default Header
