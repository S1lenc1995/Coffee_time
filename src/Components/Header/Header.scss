.header {
    background: url(./icons/sustainable_perfecti.jpg) center/cover no-repeat;
    background-attachment: fixed;
    position: relative;
     z-index: 5; 
     box-shadow: rgba(0, 0, 0, 0.3) 0px 19px 38px, rgba(0, 0, 0, 0.22) 0px 15px 12px;

    &__name {
        max-width: 1200px;
        margin: auto;
        padding-top: 200px;
        padding-bottom: 200px;
        font-family: Roboto, sans-serif;
        font-weight: 400;
        font-size: 200px;
        text-transform: uppercase;
        color: #fff;
        text-align: center;
    }

    &__nav {
        height: 100px;
        align-items: center;
        background: rgba(0, 0, 0, 0.4);
        box-shadow: 0px 4px 4px rgb(0 0 0 / 25%);
        margin-bottom: 200px;
        position: fixed;
        width: 100%;

        &--item {
            display: flex;
            justify-content: space-around;
            align-items: center;
            max-width: 1200px;
            height: 100px;
            margin: auto;

            &--icon {
                display: flex;
                justify-content: space-between;
                width: 250px;
                

                &--basket {
                  
                    position: relative;
                }
                &--favorites{    
                    position: relative;
                }
            }
        }
    }
}

.count-basket,
.count-favorites {
    position: absolute;
    top: -5px;
    right: -7px;
    font-size: 10px;
    font-size: 15px;
}


.icon-page{
    color: white;
    fill: white;
  }	

  .active{
    color: aqua;
    fill: aqua;
}

a{
    text-decoration: none;
}