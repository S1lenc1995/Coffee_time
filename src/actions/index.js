import {createAction, createAsyncThunk} from "@reduxjs/toolkit"
import { sendRequest } from "../../../hw6/src/Helpers/getData"



export const actionToggleModalAddtoBasket = createAction('ACTION_TOGGLE_MODAL_ADD_TO_BASKET')
export const actionToggleModalRemoveFromBasket = createAction('ACTION_TOGGLE_MODAL_REMOVE_FROM_BASKET')
export const actionCurrentMachine = createAction('ACTION_CURENT_MACHINE')
export const actionOrderData = createAction('ACTION_ORDER_DATA')
export const actionOrderProductData = createAction('ACTION_ORDER_PRODUCT_DATA')
export const actionCleanBasketAfterBuy = createAction('ACTION_CLEAN_BASKET_AFTER_BUY') 



export const actionFetchCoffeeMachines = createAsyncThunk("coffeeMachines/fetchData", 
    async ()=>{
       const result = await sendRequest('./data.json')
       return result
})

export const actionFetchBasketPage = createAsyncThunk("basketPage/fetchData", 
    async ()=>{
       const result = await (JSON.parse(localStorage.getItem('basket')) || [])
       return result
})

export const actionFetchFavoritesPage = createAsyncThunk("favoritesPage/fetchData", 
    async ()=>{
       const result = await (JSON.parse(localStorage.getItem('favorites')) || [])
       return result
})



