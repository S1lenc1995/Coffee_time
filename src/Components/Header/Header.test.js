import Header from "./Header";
import {render, screen} from '@testing-library/react'
import {BrowserRouter} from 'react-router-dom';
import { actionToggleModalAddtoBasket } from "../../actions";

describe("Test Header", ()=>{
    test('Snapshot Headr', ()=>{
        const header = render(<BrowserRouter><Header/></BrowserRouter>)

        expect(header).toMatchSnapshot()
    })

    test('check text', ()=>{
        render(<BrowserRouter><Header/></BrowserRouter>)

        expect(screen.getByText(/COFFEE/i))
    })
})