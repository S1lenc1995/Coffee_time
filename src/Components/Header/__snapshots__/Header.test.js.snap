// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header Snapshot Headr 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="header__nav"
        >
          <div
            class="header__nav--item"
          >
            <img
              alt="photo"
              src="https://static.tildacdn.com/tild3937-3261-4630-b235-343635633134/logo_white_mini.png"
            />
            <div
              class="header__nav--item--icon"
            >
              <div>
                <a
                  activeclassname="active"
                  aria-current="page"
                  class="icon-page active"
                  href="/"
                >
                  <p>
                    Main Page
                  </p>
                </a>
              </div>
              <div
                class="header__nav--item--icon--basket"
              >
                <a
                  activeclassname="active"
                  class="icon-page"
                  href="/basket"
                >
                  <span
                    class="count-basket"
                  />
                  <svg>
                    basketIcon.svg
                  </svg>
                </a>
              </div>
              <div
                class="header__nav--item--icon--favorites"
              >
                <a
                  activeclassname="active"
                  class="icon-page"
                  href="/favorites"
                >
                  <span
                    class="count-favorites"
                  />
                  <svg>
                    favoriteIcon.svg
                  </svg>
                </a>
              </div>
            </div>
          </div>
        </div>
        <h1
          class="header__name"
        >
          COFFEE 
          <br />
           TIME
        </h1>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="header__nav"
      >
        <div
          class="header__nav--item"
        >
          <img
            alt="photo"
            src="https://static.tildacdn.com/tild3937-3261-4630-b235-343635633134/logo_white_mini.png"
          />
          <div
            class="header__nav--item--icon"
          >
            <div>
              <a
                activeclassname="active"
                aria-current="page"
                class="icon-page active"
                href="/"
              >
                <p>
                  Main Page
                </p>
              </a>
            </div>
            <div
              class="header__nav--item--icon--basket"
            >
              <a
                activeclassname="active"
                class="icon-page"
                href="/basket"
              >
                <span
                  class="count-basket"
                />
                <svg>
                  basketIcon.svg
                </svg>
              </a>
            </div>
            <div
              class="header__nav--item--icon--favorites"
            >
              <a
                activeclassname="active"
                class="icon-page"
                href="/favorites"
              >
                <span
                  class="count-favorites"
                />
                <svg>
                  favoriteIcon.svg
                </svg>
              </a>
            </div>
          </div>
        </div>
      </div>
      <h1
        class="header__name"
      >
        COFFEE 
        <br />
         TIME
      </h1>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
